### DISCUSSION TITLE
User org filter with internal data source

### DISCUSSION THREAD

**Discussion by DavidPIsaac** [Budibase/budibase#4044] [Help] [answered]
Can I display only data related to a user's organization using the internal database and nested data providers?  Or do I have to switch to an SQL data source?

I have related tables for Users, Organizations, and Organization Records.  The extra hop through organizations means the example for showing only a user's records won't work, and the internal database doesn't support joins or custom queries.  If it's possible I should be able to figure out how to do it, but I don't want to beat my head against a wall if it is not possible.  Thanks!

**Comment by mjashanks** [comment, marked as answer]
It's not currently possible using relationships.

However you could achieve this using a "Soft" relationship, whereby you have
- User : Organisation column - a **text** field like "Acme"
- Org Record: Organisation column - a **text** field like "Acme"

Then you can filter by doing `Org Record.Organisation Equals Current User.Organisation`

Or you could use SQL!

This is something that we are looking into at the moment.

**Comment by joebudi** [comment]
Hi @DavidPIsaac  
I'm the co-founder of Budibase. Would you be interested in jumping on a quick call to discuss how you are using Budibase, and any feedback you may have? This would mean the world to us, and help us build a better product 👍 .
https://calendly.com/budibase/joe
Cheers,
Joe
